### PR TITLE
Feature/v2.0

### PR DESCRIPTION
**Pull request by vjsaini75**
**Description of your changes**

Category:

- [x] New Feature
- [ ] Bugfix
- [ ] Documentation update

Checklist:

- [ ] I have bumped all relevant version numbers
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have documented my changes if this is needed
- [ ] I have described my changes in the "description of your changes" field
- [ ] I have tested this change on a running cluster 
- [ ] I have created/corrected automated tests related to the changes, if possible/necessary
- [ ] I have checked that new and existing unit tests pass locally with my changes
- [ ] I have squashed commits if necessary

### CODE REVIEW

File: chart/values.yaml [hunk 1]
```diff
@@ -3,11 +3,12 @@
 # Declare variables to be passed into your templates.
 replicaCount: 1
 
+
 #TEMPLATE INFO: Remember to fix the below repo path and remove this comment
```
Comment (swedchef) [inline chart/values.yaml:7]:
```suggestion
```

File: app/main.py [hunk 2]
```diff
@@ -32,37 +37,44 @@ def clean_file(file_loc: str) -> pd.DataFrame:
...
+        log.info(f"file exists as : {filepath_csv} ")
+    except Exception as e:
+        log.info(f" no file exists as : {filepath_csv} ")
+        log.exception(f" reading file Failed with the message: '{e}'")
+        raise e
 
-    print("file read time in sec=", cycle_time)
-    print("database expose=", database_expose)
+    log.info(f"file read time in sec: {cycle_time}")
+    log.info(f"database expose : {database_expose} ")
```
Comment (swedchef) [inline app/main.py:77]:
```suggestion
```
You log this twice - here and line 54 - one of them should be removed.

File: chart/templates/deployment.yaml [hunk 3]
```diff
@@ -28,6 +28,16 @@ spec:
...
+          env:
+            - name: file_name
+              value: {{ .Values.file.name }}
+            - name: database_expose
+              value: {{ .Values.database.expose }}
+            - name: cycle_time
+              value: {{ .Values.cycle.time }}
+            - name: use_mock_data
+              value: {{ .Values.use.mock.data }}
+
```
Comment (swedchef) [inline chart/templates/deployment.yaml:40]:
```suggestion
          env:
            - name: cycle_time
              value: {{ .Values.cycle_time_s }}
            {{- if .Values.file_name }}
            - name: file_name
              value: {{ .Values.file_name | quote  }}
            {{- end }}
            {{- if .Values.database_expose }}
            - name: database_expose
              value: {{ .Values.database_expose | quote  }}
            {{- end }}
            {{- if .Values.use_mock_data }}
            - name: use_mock_data
              value: {{ .Values.use_mock_data | quote  }}
            {{- end }}
              
```
If you change the naming as i suggest then change it here as well. I added the "quote" as it needs it in my experience. Furthermore by doing like this they are only set if they are in the values-file.

File: tests/valid-testdata/dlr_mrid_PROD.csv [hunk 4]
```diff
@@ -1,4 +1,2 @@
 TERMINAL_EMSNAME,FAR_NEAR,AMPS_MRID,LINESEGMENT_MRID,DLR_ENABLE
-----------------,--------,---------,----------------,----------
-Dummy_line,N,12345678-1234-abcd-efgh-123456789abc,12345679-4567-ijkl-5678-123456789abc,NO
-Dummy_line,F,12345678-1234-abcd-efgh-123456789def,12345679-4567-ijkl-5678-123456789abc,YES
+----------------,--------,---------,----------------,----------
```
Comment (swedchef) [inline tests/valid-testdata/dlr_mrid_PROD.csv:2]: I don't get this - is this just an empty file?

**Review by swedchef — CHANGES_REQUESTED** [review]
Looking great - a few general comments:

1: Remember to lint - when you run the linter (flake8) it will show you some of the errors i have found as well (and a few more in other files).

2: I think we need to test the HELM asap. I am not certain it works - but i have given it my best shot.

3: When looking through my suggestions you can batch the ones you agree with and include them in a single commit - this is better than editing everything yourself if you agree (faster and less chance of making a new error). The changes you don't like - just keep them out or do those manually.

File: tests/valid-testdata/dlr_mrid_PROD.csv (hunk 4 above)
Comment (vjsaini75) [reply tests/valid-testdata/dlr_mrid_PROD.csv:2]: yes, just to avoid if no data there

File: chart/Chart.yaml [hunk 5]
```diff
@@ -1,4 +1,10 @@
...
 name: dlr-mrid
 description: A Helm chart for energinet-singularity/dlr-mrid
-version: 0.0.11
+version: 0.0.12
+
+dependencies:
+  - name: file-mover
+    repository: "https://energinet-singularity.github.io/helm-registry/"
+    version: 1.1.2
+
```
Comment (swedchef) [inline chart/Chart.yaml:10]:
```suggestion

```
Woops - your last line has empty space in it!

File: chart/templates/dlr-mrid-pvc.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: dlr-mrid-pvc
+spec:
+  accessModes:
+    - {{ .Values.dlrmridDataVolume.accessMode }}
+  resources:
+    requests:
+      storage: {{ .Values.dlrmridDataVolume.size }}
+  storageClassName: {{ .Values.dlrmridDataVolume.storageClass}}
+
```
Comment (swedchef) [inline chart/templates/dlr-mrid-pvc.yaml:12]:
```suggestion

```
Whoops - more empty space! ;)

**Review by swedchef — APPROVED** [review]
Looking great! -Think we need to clear the empty space in the lines though!

**Pull request merged**